Adds command `teams meeting transcript list`. Closes #3901 (#4206)
pnp/cli-microsoft365: docs/docs/cmd/teams/meeting/meeting-transcript-list.md

# teams meeting transcript list

> Lists all transcripts for a given meeting.

Reviewer: ✅

## Usage

```sh
m365 teams meeting transcript list [options]
```

## Options

`-u, --userId [userId]`

> : The id of the user, omit to list attendance reports for current signed in user. Use either  `id`, `userName` or `email`, but not multiple.

Reviewer: good catch

> : The id of the user, omit to list attendance reports for current signed in user. Use either  `id`, `userName` or `email`, but not multiple.

Reviewer: ✅

> `--upn [upn]`

Reviewer 2: Instead of upn, this should be `userName` as mentioned in the specs. I think you misunderstood my last remark 😅. I meant to clarify the description, not change the option name 😄

Reviewer: 🤦‍♂️

Reviewer: @milanholemans not that I don't want to revert it 😉 but TBH what do you think we actually leave this `upn`? I mean if it was not clear what `userName` was and we need to clearly state it in the description why not just rename it to `upn` so what I did? I mean it may not be clearer than that right 🙂🤔
also we already use `upn` options in some `spo` commands so it would not be something totally new I guess 🤔

Reviewer 2: I'm not the biggest fan of `userName` as option, but this is something we use almost everywhere. If you check the other `teams meeting` commands, they all use `userName`, same goes to `aad user` commands etc. So for the sake of consistency, I'd use here `userName` as well.

Reviewer: ok if it will be consistent with other teams commands then I agree 👍
I wonder if we should align and use either `userName` or `upn` everywhere (as I know the second is used in many places on `spo`)  but this is a different story (issue) 😉

Reviewer 2: Seems like 39 commands use `userName` and 10 use `upn`. Aligning these would be a good idea indeed. Consistency is quite important.

Reviewer: I agree 😉 
so on whose 'todo' list is it now 🤔 Yours or mine 😅😜?

Reviewer 2: The honor is yours 😄

Reviewer: but by all means, I do not whish to steal that honor from you.  😅

Reviewer: ✅

> : The name of the user, omit to list attendance reports for current signed in user. Use either `id`, `userName` or `email`, but not multiple.

Reviewer: sure

> : The name of the user, omit to list attendance reports for current signed in user. Use either `id`, `userName` or `email`, but not multiple.

Reviewer: ✅, it is very misleading what is the difference between upn and email 😅. I even checked that in some cases we are using email as upn which unfortunately as I investigated is not always the same.
so for example in `FileRoleAssignmentAdd` we use `SpoUserGetCommand` where we do 
```
const userGetCommandOptions: SpoUserGetCommandOptions = {
        webUrl: options.webUrl,
        email: options.upn, // <--- not 100% clean and safe I guess 
        id: undefined,
        output: 'json',
        debug: this.debug,
        verbose: this.verbose
      };
      ```
      
      ok but anyway.. done

> : The name of the user, omit to list attendance reports for current signed in user. Use either `id`, `userName` or `email`, but not multiple.

Reviewer 2: Upn and Email is indeed not always the same.
If I recall correctly, you can change email without any problem. Changing upn will break stuff in M365.

`--email [email]`

> : The email of the user, omit to list attendance reports for current signed in user. Use either `id`, `userName` or `email`, but not multiple.

Reviewer: sorry for that, TBH I didn't put too much of an attention to docs 😉

`-m, --meetingId <meetingId>`

> : The Id of the meeting.

Reviewer: ✅

--8<-- "docs/cmd/_global.md"

## Examples

Lists all transcripts made for the current signed in user and Microsoft Teams meeting with given id

```sh
m365 teams meeting transcript list --meetingId MSo1N2Y5ZGFjYy03MWJmLTQ3NDMtYjQxMy01M2EdFGkdRWHJlQ
```

> Lists all transcripts made for the _[garthf@contoso.com](mailto:garthf@contoso.com)_ and Microsoft Teams meeting with given id

Reviewer 2: I don't think it's very useful to make a mailto link for a fake email.

```suggestion
Lists all transcripts for a meeting of a specific user
```

Reviewer: ✅

```sh
m365 teams meeting transcript list --userName garthf@contoso.com --meetingId MSo1N2Y5ZGFjYy03MWJmLTQ3NDMtYjQxMy01M2EdFGkdRWHJlQ
```

## Remarks

!!! attention
    This command is based on a Microsoft Graph API that is currently in preview and is subject to change once the API reached general availability.

> 

Reviewer 2: Added an extra remark that when you are using application permissions, you should create and extra access policy.

## Response

=== "JSON"

    ```json
    [
      {
          "id": "MSMjMCMjZDAwYWU3NjUtNmM2Yi00NjQxLTgwMWQtMTkzMmFmMjEzNzdh",
          "meetingId": "MSpiZTExZjUyMy0yYTRkLTRlYWUtOWQ0Mi0yNzc0MTA4OTNjNDEqMCoqMTk6bWVldGluZ19aakU0WmpVMllqY3RZMkV3T1MwME1UaGtMV0prWlRRdE1qRXhPVGN4T0RaalpUUTJAdGhyZWFkLnYy",
          "meetingOrganizerId": "be11f523-2a4d-4eae-9d42-277410893c41",
          "transcriptContentUrl": "https://graph.microsoft.com/beta/users/be11f523-2a4d-4eae-9d42-277410893c41/onlineMeetings/MSpiZTExZjUyMy0yYTRkLTRlYWUtOWQ0Mi0yNzc0MTA4OTNjNDEqMCoqMTk6bWVldGluZ19aakU0WmpVMllqY3RZMkV3T1MwME1UaGtMV0prWlRRdE1qRXhPVGN4T0RaalpUUTJAdGhyZWFkLnYy/transcripts/MSMjMCMjZDAwYWU3NjUtNmM2Yi00NjQxLTgwMWQtMTkzMmFmMjEzNzdh/content",
          "createdDateTime": "2021-09-17T06:09:24.8968037Z"
      }
    ]
    ```

=== "Text"

    ```text
    id                                                        createdDateTime
    ------------------------------------                      ---------------------
    MSMjMCMjZDAwYWU3NjUtNmM2Yi00NjQxLTgwMWQtMTkzMmFmMjEzNzdh  2021-09-17T06:09:24.8968037Z
    ```

=== "CSV"

    ```csv
    id,createdDateTime

>     MSMjMCMjZDAwYWU3NjUtNmM2Yi00NjQxLTgwMWQtMTkzMmFmMjEzNzdh,2021-09-17T06:09:24.8968037Z

Reviewer 2: If you rebase with the latest main, then you'll see that CSV has more properties now.

Reviewer: ✅

> MSMjMCMjMTAxOGIzZDgtMWJlMy00Y2Y2LWE4YjUtODFhNmVhYzFjNTYz,MSpiZTExZjUyMy0yYTRkLTRlYWUtOWQ0Mi0yNzc0MTA4OTNjNDEqMCoqMTk6bWVldGluZ19aakU0WmpVMllqY3RZMkV3T1MwME1UaGtMV0prWlRRdE1qRXhPVGN4T0RaalpUUTJAdGhyZWFkLnYy,be11f523-2a4d-4eae-9d42-277410893c41,https://graph.microsoft.com/beta/users/be11f523-2a4d-4eae-9d42-277410893c41/onlineMeetings/MSpiZTExZjUyMy0yYTRkLTRlYWUtOWQ0Mi0yNzc0MTA4OTNjNDEqMCoqMTk6bWVldGluZ19aakU0WmpVMllqY3RZMkV3T1MwME1UaGtMV0prWlRRdE1qRXhPVGN4T0RaalpUUTJAdGhyZWFkLnYy/transcripts/MSMjMCMjMTAxOGIzZDgtMWJlMy00Y2Y2LWE4YjUtODFhNmVhYzFjNTYz/content,2023-03-25T21:32:08.5586288Z

Reviewer: ✅
